pkg/apis/componentconfig/v1alpha1/temp_v1a1_refactor.gogo

```diff
@@ -0,0 +1,548 @@
+  // lockFilePath is the path that kubelet will use to as a lock file.
+  // It uses this file as a lock to synchronize with other kubelet processes
+  // that may be running.
+  LockFilePath *string `json:"lockFilePath"`
+  // ExitOnLockContention is a flag that signifies to the kubelet that it is running
+  // in "bootstrap" mode. This requires that 'LockFilePath' has been set.
+  // This will cause the kubelet to listen to inotify events on the lock file,
+  // releasing it and exiting when another process tries to open that file.
+  ExitOnLockContention bool `json:"exitOnLockContention"`
+
+  // rootDirectory is the directory path to place kubelet files (volume
+  // mounts,etc).
+  RootDirectory string `json:"rootDirectory"`
+
+  // containerRuntime is the container runtime to use.
+  ContainerRuntime string `json:"containerRuntime"`
+  // remoteRuntimeEndpoint is the endpoint of remote runtime service
+  RemoteRuntimeEndpoint string `json:"remoteRuntimeEndpoint"`
+  // remoteImageEndpoint is the endpoint of remote image service
+  RemoteImageEndpoint string `json:"remoteImageEndpoint"`
+  // runtimeRequestTimeout is the timeout for all runtime requests except long running
+  // requests - pull, logs, exec and attach.
+  RuntimeRequestTimeout metav1.Duration `json:"runtimeRequestTimeout"`
+}
+
+/* BEGIN RUNTIME REMOVAL */
+
+// I believe the plan is to remove ALL of these from the Kubelet/Node configuration
+// e.g. they become flags-only for now and are immediately marked deprecated, for later removal.
+
+// Note: The WIP PR for making these flags-only is #44061
+
+type RuntimeOptions struct {
+  // networkPluginName is the name of the network plugin to be invoked for
+  // various events in kubelet/pod lifecycle
+  NetworkPluginName string `json:"networkPluginName"`
+  // networkPluginDir is the full path of the directory in which to search
+  // for network plugins (and, for backwards-compat, CNI config files)
+  NetworkPluginDir string `json:"networkPluginDir"`
+  // networkPluginMTU is the MTU to be passed to the network plugin,
+  // and overrides the default MTU for cases where it cannot be automatically
+  // computed (such as IPSEC).
+  NetworkPluginMTU int32 `json:"networkPluginMTU"`
+  // CNIConfDir is the full path of the directory in which to seardch for
+  // CNI config files
+  CNIConfDir string `json:"cniConfDir"`
+  // CNIBinDir is the full path of the directory in which to search for
+  // CNI plugin binaries
+  CNIBinDir string `json:"cniBinDir"`
+  // podInfraContainerImage is the image whose network/ipc namespaces
+  // containers in each pod will use.
+  PodInfraContainerImage string `json:"podInfraContainerImage"`
+}
+
+type DockerOptions struct {
+  // dockerEndpoint is the path to the docker endpoint to communicate with.
+  DockerEndpoint string `json:"dockerEndpoint"`
+  // enable gathering custom metrics.
+  EnableCustomMetrics bool `json:"enableCustomMetrics"` // for Docker
```

Comment:
Don't think `EnableCustomMetrics` belong in the `DockerOptions`. We support this through cadvisor, and may eventually move this out of kubelet.